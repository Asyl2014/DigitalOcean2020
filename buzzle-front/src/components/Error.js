import React from 'react';
import { connect } from 'react-redux';
import { clearError } from '../actions/error';

function Error(props) {
    const error = props.error; 
    const clearError = props.clearError;

    
    return ( error ?    
        <div className="row justify-content-center mt-3">
            <div className="col-md-4">
                <div className="alert alert-danger">
                    {error}
                    <button onClick={clearError} type="button" class="close">
                        <span>&times;</span>
                    </button>
                </div>
            </div>
        </div>
        :
        <></>               
    );
}
const mapStateToProps = state => ({
    'error': state.error
});

const mapDispatchToProps = dispatch => ({
    'clearError': () => dispatch(clearError())
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Error);
