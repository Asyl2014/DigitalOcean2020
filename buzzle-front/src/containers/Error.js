<<<<<<< HEAD
import { connect } from 'react-redux';



import { clearError } from '../actions/error.js';
import Error from '../components/Error.js';

const mapStateToProps = state => ({
   'error': state.error
});

const mapDispatchToProps = dispatch => ({
    'clearError': () => dispatch(clearError())
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Error);
=======
import { connect } from 'react-redux';

import { clearError } from '../actions/error.js';
import Error from '../components/Error.js';

const mapStateToProps = state => ({
    'error': state.error
});

const mapDispatchToProps = dispatch => ({
    'clearError': () => dispatch(clearError())
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Error);
>>>>>>> 2667221132704740c4ff83a8469713a301774162
