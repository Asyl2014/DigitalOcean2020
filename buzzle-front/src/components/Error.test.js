import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import Error from './Error.js';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Error />, div);
});

it('renders without crashing', () => {
  const wrapper = shallow(<Error error="Something went wrong..."/>);
  expect(wrapper.contains("Something went wrong...")).toBe(true);
});