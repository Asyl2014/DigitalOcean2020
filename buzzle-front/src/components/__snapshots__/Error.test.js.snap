// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and equal to snapshot 1`] = `
<div
  className="row justify-content-center mt-3"
>
  <div
    className="col-md-4"
  >
    <div
      className="alert alert-danger"
    >
      Something  went wrong...
    </div>
  </div>
</div>
`;
